test/run.sh

```diff
@@ -153,7 +156,7 @@ rm -rf "$TMP" \
 
 mkdir -p "$TMP"
 
-if [ "$1" == "record" ]
+if [ "$SUBCOMMAND" == "record" ]
```

Comment:
Instead of using the potentially unbound `$1`, use the bound-to-be-bound `$SUBCOMMAND`, which also reads easier.